### DISCUSSION TITLE
sampling over partitions?

### DISCUSSION THREAD

**Discussion by cottrell** [duckdb/duckdb#3644] [Q&A]
Is it possibly to do something to sample over partitions using SAMPLE or TABLESAMPLE? 

https://duckdb.org/docs/sql/samples

The standard way works but I think can be quite slow and isn't taking advantage of SAMPLE or TABLESAMPLE which I think have various "local" optimizations? The standard way being:
```
select * from (select *, rank() over (partition by my_part order by random()) from data) where rank <= 100
```
https://www.redpill-linpro.com/techblog/2021/05/07/getting-random-rows-faster.html

**Comment by Alex-Monahan** [comment]
Hello!

I'm not exactly sure what you mean by sampling over partitions! However, `system` sampling will sample per vector and should be a very fast way to sample!

**Reply by cottrell** [reply]
I think I need reservoir as I want deterministic samples. 

And by sampling over partitions I mean something like max 1000 samples per date for examples. 

Or 1000 samples per country. 

Things like that.
